pkg/apis/openebs.io/upgrade/v1alpha1/upgradetask.go

```diff
@@ -0,0 +1,201 @@
+	// CStorPool contains the details of the cstor pool to be upgraded
+	CStorPool *CStorPool `json:"cstorPool,omitempty" protobuf:"bytes,3,opt,name=cstorPool"`
+	// StoragePoolClaim contains the details of the storage pool claim to be upgraded
+	StoragePoolClaim *StoragePoolClaim `json:"storagePoolClaim,omitempty" protobuf:"bytes,4,opt,name=storagePoolClaim"`
+}
+
+// JivaVolume is the ResourceType for jiva volume
+type JivaVolume struct {
+	// PVName contains the name of the pv associated with the jiva volume
+	PVName string `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	// Options can be used to change the default behaviour of upgrade
+	Options *ResourceOptions `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorVolume is the ResourceType for cstor volume
+type CStorVolume struct {
+	// PVName contains the name of the pv associated with the cstor volume
+	PVName string `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	// Options can be used to change the default behaviour of upgrade
+	Options *ResourceOptions `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorPool is the ResourceType for cstor pool
+type CStorPool struct {
+	// PoolName contains the name of the cstor pool to be upgraded
+	PoolName string `json:"poolName,omitempty" protobuf:"bytes,1,name=poolName"`
+	// Options can be used to change the default behaviour of upgrade
+	Options *ResourceOptions `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// StoragePoolClaim is the ResourceType for storage pool claim
+type StoragePoolClaim struct {
+	// SPCName contains the name of the storage pool claim to be upgraded
+	SPCName string `json:"spcName,omitempty" protobuf:"bytes,1,name=spcName"`
+	// Options can be used to change the default behaviour of upgrade
+	Options *ResourceOptions `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// ResourceOptions provides additional options for a particular resource
+type ResourceOptions struct {
+	// IgnoreStepsOnError allows to ignore steps which failed
+	IgnoreStepsOnError []string `json:"ignoreStepsOnError,omitempty" protobuf:"bytes,1,opt,name=ignoreStepsOnError"`
+}
+
+// UpgradeTaskStatus provides status of a upgradeTask
+type UpgradeTaskStatus struct {
+	// Phase indicates if a upgradeTask is started, success or errored
+	Phase UpgradePhase `json:"phase,omitempty" protobuf:"bytes,1,opt,name=phase,casttype=UpgradePhase"`
+	// StartTime of Upgrade
+	StartTime metav1.Time `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+	// CompletedTime of Upgrade
+	CompletedTime metav1.Time `json:"completedTime,omitempty" protobuf:"bytes,3,opt,name=completedTime"`
+	// UpgradeDetailedStatuses contains the list of statuses of each step
+	UpgradeDetailedStatuses []UpgradeDetailedStatuses `json:"upgradeDetailedStatuses,omitempty" protobuf:"bytes,4,rep,name=upgradeDetailedStatuses"`
+}
```

Comment:
@shubham14bajpai @AmitKumarDas Should we have a retries field here so that we can know the no of retries or the no of times the pod has run. It will also help to know the top level operators that the job has finished its execution.

Reply:
@sagarkrsd added the `Retries` field